### PR TITLE
SMOK-43280 : Shotgun toolkit was returning an error when client imple…

### PR DESCRIPTION
**Pull request by mtiudaeu**
…ment there own version of getCustomExport hook. We cannot assume any values of the dictionnary will exist.

### CODE REVIEW

File: flame_hooks/sg_export_hook.py [hunk 1]
```diff
@@ -56,7 +56,10 @@ def preCustomExport(info, userData):
         return
 
     # get the preset that the user selected from the menu
-    current_preset = userData.get("preset_title")
+    current_preset = userData.get("sg_preset_title")
```
Comment (jfboismenu) [inline flame_hooks/sg_export_hook.py]: Why the change from `preset_title` to `sg_preset_title`

File: flame_hooks/sg_export_hook.py [hunk 2]
```diff
@@ -56,7 +56,10 @@ def preCustomExport(info, userData):
...
+    # We can get here with a client defined hook. Ignore it if it is not shotgun related.
```
Comment (jfboismenu) [inline flame_hooks/sg_export_hook.py]: Can you provide a bit of history behind this? I'm not familiar enough with the preCustomExport hook. Why is there a value in the userData that Flame passes in that is Shotgun specific?

Comment (mtiudaeu) [inline flame_hooks/sg_export_hook.py]: How it works is when you right-click on a clip in flame, the python hook getCustomExportProfiles gets called to populate the right-click menu. So shotgun toolkit getCustomExportProfiles will get call and also the getCustomExportProfiles defined in /opt/Autodesk/flame_2018/python/export_hook.py (or another custom path) will get called. Depending on which of the UI button the user click, preCustomExport userData you will received will be from the shotgun hook, or the client's specific hook. That's why you cannot assume the value of the dictionary.
We have no plugin manager in flame and some collision can happen between the client's hooks and shotgun's hooks. That's a limitation of the python hook.

File: flame_hooks/sg_export_hook.py (hunk 1 above)
Comment (mtiudaeu) [inline flame_hooks/sg_export_hook.py]: I figured adding "sg_" reduce the risk of collision with a client's implementation of preCustomExport. It's minor, it doesn't really matter, I don't mind either way.

Comment (jfboismenu) [inline flame_hooks/sg_export_hook.py]: Ah, I get it now!

File: flame_hooks/sg_export_hook.py (hunk 2 above)
Comment (jfboismenu) [inline flame_hooks/sg_export_hook.py]: Got it. Thanks.
